presto-main/src/main/java/com/facebook/presto/operator/MergeJoinOperatorFactory.java

```diff
@@ -0,0 +1,93 @@
+{
+    private final int operatorId;
+    private final PlanNodeId planNodeId;
+    private final List<Type> leftTypes;
+    private final List<Type> leftOutputTypes;
+    private final List<Integer> leftOutputChannels;
+    private final List<Type> rightTypes;
+    private final List<Type> rightOutputTypes;
+    private final List<Integer> rightOutputChannels;
+    private final List<Integer> leftJoinChannels;
+    private final List<Integer> rightJoinChannels;
+    private final JoinNode.Type joinType;
+    private final MergeJoinSourceManager mergeJoinSourceManager;
+    private boolean closed;
+
+    public MergeJoinOperatorFactory(
+            int operatorId,
+            PlanNodeId planNodeId,
+            MergeJoinSourceManager mergeJoinSourceManager,
+            List<Type> leftTypes,
+            List<Type> leftOutputTypes,
+            List<Integer> leftOutputChannels,
+            List<Type> rightTypes,
+            List<Type> rightOutputTypes,
+            List<Integer> rightOutputChannels,
+            List<Integer> leftJoinChannels,
+            List<Integer> rightJoinChannels,
+            JoinNode.Type joinType)
+    {
+        this.operatorId = operatorId;
+        this.planNodeId = requireNonNull(planNodeId, "planNodeId is null");
+        this.leftTypes = ImmutableList.copyOf(requireNonNull(leftTypes, "leftTypes is null"));
+        this.leftOutputTypes = ImmutableList.copyOf(requireNonNull(leftOutputTypes, "leftOutputTypes is null"));
+        this.leftOutputChannels = ImmutableList.copyOf(requireNonNull(leftOutputChannels, "leftOutputChannels is null"));
+        this.rightTypes = ImmutableList.copyOf(requireNonNull(rightTypes, "rightTypes is null"));
+        this.rightOutputTypes = ImmutableList.copyOf(requireNonNull(rightOutputTypes, "rightOutputTypes is null"));
+        this.rightOutputChannels = ImmutableList.copyOf(requireNonNull(rightOutputChannels, "rightOutputChannels is null"));
+        this.leftJoinChannels = ImmutableList.copyOf(requireNonNull(leftJoinChannels, " is null"));
+        this.rightJoinChannels = ImmutableList.copyOf(requireNonNull(rightJoinChannels, " is null"));
+        this.joinType = requireNonNull(joinType, "joinType is null");
+        this.mergeJoinSourceManager = requireNonNull(mergeJoinSourceManager, "mergeJoinSourceManager is null");
+    }
+
+    @Override
+    public Operator createOperator(DriverContext driverContext)
+    {
+        checkState(!closed, "Factory is already closed");
+
+        OperatorContext operatorContext = driverContext.addOperatorContext(operatorId, planNodeId, MergeJoinOperator.class.getSimpleName());
+        MergeJoinSource mergeJoinSource = mergeJoinSourceManager.getMergeJoinSource(driverContext.getLifespan(), null);
+        return new MergeJoinOperator(operatorContext, leftTypes, leftOutputTypes, leftOutputChannels, rightTypes, rightOutputTypes, rightOutputChannels, leftJoinChannels, rightJoinChannels, joinType, mergeJoinSource);
```

Comment:
NIT: this line exceeds the recommended length, break the parameters into multiple lines